File: Php/1-50/12.php
<?php
$n = intval(readline());
$a = [];
for ($i = 0; $i <= $n; $i++) {
    $a[] = $i;
}

$a[1] = 0;
$i = 2;
while ($i <= $n) {
    if ($a[$i] != 0) {
        $j = $i + $i;
        while ($j <= $n) {
            $a[$j] = 0;
            $j = $j + $i;
        }
    }
    $i += 1;
}

$a = array_filter($a);
$a = array_diff($a, [0]);
$ism = ["Bobur", "Ali"];

if ($n == 1) {
    echo "Ali\n";
} else {
    echo $ism[count($a) % 2] . "\n";
}
?>

File: Kotlin/1-50/12.kt
fun main() {
    val n = readLine()!!.toInt()
    val a = IntArray(n + 1) { it }

    a[1] = 0
    var i = 2
    while (i <= n) {
        if (a[i] != 0) {
            var j = i + i
            while (j <= n) {
                a[j] = 0
                j += i
            }
        }
        i++
    }

    val primes = a.filter { it != 0 }
    val ism = arrayOf("Bobur", "Ali")

    if (n == 1) {
        println("Ali")
    } else {
        println(ism[primes.size % 2])
    }
}
